activities.el

```diff
@@ -801,6 +851,116 @@ activity's name is NAME."
                   "In the meantime, it's recommended to not use buffers of this major mode in an activity's layout; or you may simply ignore this error and use the other buffers in the activity.")
           (current-buffer)))))
 
+(defvar activities--age-spec
+  `((?Y "year"   "years"   ,(round (* 60 60 24 365.2425)))
+    (?M "month"  "months"  ,(round (* 60 60 24 30.436875)))
+    (?w "week"   "weeks"   ,(* 60 60 24 7))
+    (?d "day"    "days"    ,(* 60 60 24))
+    (?h "hour"   "hours"   ,(* 60 60))
+    (?m "minute" "minutes" 60)
+    (?s "second" "seconds" 1))
+  "Age specification.  See `magit--age-spec', which this duplicates.")
+
+(defun activities--age (age &optional abbreviate)
+  "Summarize AGE and possibly ABBREVIATE.
+Adapted from `magit--age'."
```

Comment:
I'm not sure if this can be adapted here, because Activities is in GNU ELPA and is copyrighted by the FSF, while Magit is not copyrighted by the FSF.  @tarsius Could you please advise here?

Reply:
That's correct. However, I am the only one who ever touched this function and the associated variable, and would be happy to contribute this to Emacs. I would ask @jdtsmith to do the work of getting it integrated (or, if that's what the maintainers request, replaced by a better implementation).

Reply:
@tarsius Thanks.  With your being the author, if you consent, then, can it just be "contributed to Emacs" via this package, since ELPA packages are "part of GNU Emacs"?

Reply:
Well yes, but let's wait and see if @jdtsmith would be willing to get it into Emacs first. Doesn't have to be this implementation, but something like this would certainly be useful in more than these two packages.

As for the used algorithm, one might want to consider using another threshold than 1. I don't really mind that 2.0 and 2.49 years are both "2 years", but when it comes to 0.51 and 1.49 both being "1 year", I feel a bit differently. Maybe this should be customizable, maybe even per time unit.

If I remember correctly, I did shop around for other implementations before writing mine, but the once I found all addressed this by using more than one time unit, which doesn't work in our limited space constraint.

Reply:
Happy to help out.  What did you envision?  I also use some color and date-related functionality from `vc-annotate`, perhaps it could go there (though I rarely use `vc` anymore, because magit ;).  

A little more granularity would be nice; maybe a unicode ½ could enter.

Reply:
> A little more granularity would be nice; maybe a unicode ½ could enter.

My 2 cents: Sure ½ years is nice, but 12½ years is unnecessary "precision". I like this progression of half year steps better 6 months, 12 months, 18 months, 2 years, 3 years (for 2.5), 3 years (for 3), and so on. This is actually more precise (in the range were it matters) than the ½ approach: 4/6/8 months vs. ½/½/½ years. And that's what I don't like about the half-step approach, it suggests precision that just isn't there.

Reply:
The progression would go 1 hour, 1½ hours, 1 day, 1½ days, 1 week, 1½ weeks, 1 month, 1½ months, 6 months, 1 year, 1½ years.  I can see an argument for dropping the ½ and reverting to integer rounding after a certain base value, perhaps 3½?  What about 7½ months vs. 7 or 8 months?  I could go either way on that one.  Most relevant for years (I briefly toyed with adding "decades" but... no).  I agree that 29½ years is fairly silly.   The current `seconds-to-string` uses 1/100ths across the board btw.

Reply:
https://debbugs.gnu.org/cgi/bugreport.cgi?bug=71572

Reply:
I think I am going to stick what I have now.

> The current seconds-to-string uses 1/100ths across the board btw.

Had I known that function and if it used 1/10, I probably would have used it.

A code comment mentions that there is a fancier implementation to be found in `mastodon.el`. I wasn't able to locate that, could you?

Reply:
No I didn't.  Take a look at the patch I submitted (should also be in your inbox).  Happy for further  input.

Reply:
There obviously are different approaches to this task. For the reasons given, I don't like the half-step approach much, and looking at the implementation did not change that. That shouldn't keep you from iterating on it. There's room for different approaches. If something lands in Emacs, it would probably be a good idea, if it were done in a way that allowed for switching in an alternative implementation. (I.e., using a `...-function` variable, though in the absence of that, an `:override` advice would of course also work.)

Reply:
A good design goal IMO is "something tarsius would be happy to sub in for magit--age at some point", so feel free to express your ideas [there](https://debbugs.gnu.org/cgi/bugreport.cgi?bug=71573).  So far no response.  `HALF` is optional btw, default off.